tests/php/src/ReaderThemeLoaderTest.php

```diff
@@ -27,7 +25,8 @@ final class ReaderThemeLoaderTest extends WP_UnitTestCase {
 
 	public function setUp() {
 		parent::setUp();
-		$this->instance = new ReaderThemeLoader();
+		remove_all_filters( 'wp_prepare_themes_for_js' );
```

Comment:
@schlessera This ended up being the key to fixing the conflict between `\AmpProject\AmpWP\Tests\ReaderThemeLoaderTest::test_filter_wp_prepare_themes_to_indicate_reader_theme` and `Test_AMP_Template_Customizer::test_get_active_theme_import_settings`, but I'm not entirely sure why.

Reply:
Something about either the `ReaderThemeLoader` or its tests is flaky.

I disabled the above `remove_all_filters()` to see what happens.

Running `ReaderThemeLoaderTest` class only => tests pass.
Running entire test suite => tests fail on the assert for the `delete` action.
Running tests with `--filter="Reader"` => tests fail with `Undefined index: REQUEST_URI`

Reply:
Based on the code where it fails, I have the suspicion that it could have to do with user capabilities. Are these reset automatically by the WP test case?

Reply:
It seems this was a side-effect of abusing the `Services` helper to retrieve the objects to test.

Contrary to the injector, the service container immediately does call a `register()` method on any `Service` that is marked as `Registerable`.

I've changed things a bit in how the `Services` helper works and how the `DependencyInjectionTestCase` initializes it. However, we should only ever use the injector, _not_ the service container within the tests, unless we have a good reason to use that container.

Reply:
Ha, and now that the tests make more sense, conceptually, I'm back to having an infinite recursion from the FileLoader, but this time in an actual test failure... So back to fixing the actual problem!

![Image 2020-09-22 at 3 13 26 PM](https://user-images.githubusercontent.com/83631/93886833-2fa15000-fce6-11ea-89d3-fea407df1e4b.png)